### PR TITLE
Add a service to handle batch searches over HTTP POST

### PR DESCRIPTION
**Pull request by trevor-m-ford**
There are two commits here:
  1. Refactor batch search CLI code, extract it for reuse in CLI and HTTP server
  2. Add function to HTTP server, call our batch search

**Test Plan**
1. Verify that CLI still works as previously:
```
% go run ./cmd/cc_batchsearch -file=/Users/trevorford/Projects/ofac/tests/post_input.csv -min-match=0.96 -threshold=0.99 -write
```
Here is the file: [post_input.csv](https://github.com/ComplyCo/watchman/files/13273612/post_input.csv)
The output file will be written to the same directory as the input file, named post_input_output.csv - or <original>_output.<original ext>

2. Run the server locally (remote coming soon!): `% make build && make run`
3. Send a POST request to `localhost:8084/search/batch`
   Add a param called "csvFile", change the type to File (please see screenshot below)

| Test case | Image |
| :--: | :--: |
| Defaults |Screenshot 2023-11-06 at 3 52 29 PM|
| fuzzy=60% | Screenshot 2023-11-06 at 3 56 46 PM |
| fuzzy=60% threshold=70% | Screenshot 2023-11-06 at 3 58 35 PM |
| sdn-type=Vessel | Screenshot 2023-11-06 at 4 12 31 PM |

### CODE REVIEW

File: cmd/internal/search_batch.go [hunk 1]
```diff
@@ -28,6 +34,73 @@ var (
...
+			http.Error(w, "Unable to write response", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func setFlagsFromFormFields(r *http.Request) {
+	if threshold := r.FormValue("threshold"); threshold != "" {
+		if f, err := strconv.ParseFloat(threshold, 64); err == nil {
+			*flagThreshold = f
```
Comment (sdemjanenko) [inline cmd/internal/search_batch.go:84]: What is this doing? Is `flagThreshold` scoped to the request?

Comment (trevor-m-ford) [reply cmd/internal/search_batch.go:84]: Good question! I should probably find a more appropriate way to represent the parameters here; in the interest of speed I just moved the command line parameters over.

Running the Watchman server locally, I ran into the problem that the values persist between calls. (e.g. send threshold=0.6 in first call, then don't send in second call = it uses 0.6)

I did this hacky approach for now. If the POST request doesn't include a parameter, set `flagX` to the default value [to immitate the CLI behavior].

I will definitely do this differently in our own OFAC code.

File: cmd/internal/search_batch.go [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+		*flagMinNameScore = 0.90
+	}
+	if sdn_type := r.FormValue("sdn-type"); sdn_type != "" {
+		*flagSdnType = sdn_type
+	} else {
+		*flagSdnType = "individual"
+	}
+	if request_id := r.FormValue("request-id"); request_id != "" {
+		*flagRequestID = request_id
+	}
```
Comment (trevor-m-ford) [inline cmd/internal/search_batch.go:103]: I realize that `flagRequestID` will suffer from the same persistence issue.
I'll fix this and update PR.

Comment (sdemjanenko) [reply cmd/internal/search_batch.go:103]: If I understand correctly, we can probably skip the request-id flag as it is only used for tracing purposes and we don't need to support that in this research code.

**Pull request merged**